File: docs/code-examples/line_strip3d_simple.cpp
// Log a simple line strip.

#include <rerun.hpp>

int main() {
    auto rec = rerun::RecordingStream("rerun_example_line_strip3d");
    rec.connect().throw_on_failure();

    rerun::components::LineStrip3D linestrip({
        {0.f, 0.f, 0.f},
        {0.f, 0.f, 1.f},
        {1.f, 0.f, 0.f},
        {1.f, 0.f, 1.f},
        {1.f, 1.f, 0.f},
        {1.f, 1.f, 1.f},
        {0.f, 1.f, 0.f},
        {0.f, 1.f, 1.f},
    });
    rec.log("strip", rerun::LineStrips3D(linestrip));
}

File: docs/snippets/all/line_strip3d_simple.py
"""Log a simple line strip."""
import rerun as rr

rr.init("rerun_example_line_strip3d", spawn=True)

points = [
    [0, 0, 0],
    [0, 0, 1],
    [1, 0, 0],
    [1, 0, 1],
    [1, 1, 0],
    [1, 1, 1],
    [0, 1, 0],
    [0, 1, 1],
]

rr.log("strip", rr.LineStrips3D([points]))
